File: src/main/java/features/reports/PdfElementData.java
package features.reports;

import com.itextpdf.layout.element.IBlockElement;
import core.data.Data;

/**
 * Host class for {@link IBlockElement} to transmit throughout framework.
 */
public class PdfElementData extends Data {
    public final IBlockElement pdfElement;

    public PdfElementData(IBlockElement pdfElement) {
        super();
        this.pdfElement = pdfElement;
    }
}

File: src/main/kotlin/com/mirf/features/reports/PdfElementData.kt
package com.mirf.features.reports

import com.itextpdf.layout.element.IBlockElement
import com.mirf.core.data.MirfData

/**
 * Host class for [IBlockElement] to transmit throughout framework.
 */
class PdfElementData(val pdfElement: IBlockElement) : MirfData()
